skimage/filters/_frangi.py

```diff
@@ -0,0 +1,195 @@
+
+    # Filtering for all sigmas
+    for i in range(len(sigmas)):
+        sigma = sigmas[i]
+        # Make 2D hessian
+        (Dxx, Dxy, Dyy) = hessian_matrix(image, sigma)
+
+        # Correct for scale
+        Dxx = (sigma ** 2) * Dxx
+        Dxy = (sigma ** 2) * Dxy
+        Dyy = (sigma ** 2) * Dyy
+
+        # Calculate (abs sorted) eigenvalues and vectors
+        (lambda1, lambda2) = hessian_matrix_eigvals(Dxx, Dxy, Dyy)
+
+        # Compute some similarity measures
+        lambda1[lambda1 == 0] = 1e-10
+        rb = (lambda2 / lambda1) ** 2
+        s2 = lambda1 ** 2 + lambda2 ** 2
+
+        # Compute the output image
+        filtered = np.exp(-rb / beta1) * (np.ones(np.shape(image)) -
+                                          np.exp(-s2 / beta2))
+
+        # Store the results in 3D matrices
+        filtered_array[i] = filtered
+        lambdas_array[i] = lambda1
+    return filtered_array, lambdas_array
+
+
+def frangi(image, scale=(1, 10), scale_step=2, beta1=0.5, beta2=15,
+           black_ridges=True):
+    """Filter an image with the Frangi filter.
+
+    This filter can be used to detect continous edges, e.g. vessels,
+    wrinkles, rivers. It can be used to calculate the fraction of the
+    whole image containing such objects.
+
+    Calculates the eigenvectors of the Hessian to compute the likeliness of
+    an image region to vessels, according to the method described in _[1].
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Array with input image data.
+    scale : tuple of floats, optional
+        The range of sigmas used.
+    scale_step : float, optional
+        Step size between sigmas.
+    beta1 : float, optional
+        Frangi correction constant.
+    beta2 : float, optional
+        Frangi correction constant.
+    black_ridges : boolean, optional
+        Detect black ridges (default) set to true, for
+        white ridges set to false.
+
+    Returns
+    -------
+    out : (N, M) ndarray
+        Filtered image (maximum of pixels across all scales).
+
+    Notes
+    -----
+    Written by Marc Schrijver, 2/11/2001
+    Re-Written by D. J. Kroon University of Twente (May 2009)
+
+    References
+    ----------
+    .. [1] A. Frangi, W. Niessen, K. Vincken, and M. Viergever. "Multiscale
+           vessel enhancement filtering," In LNCS, vol. 1496, pages 130-137,
+           Germany, 1998. Springer-Verlag.
+    .. [2] Kroon, D.J.: Hessian based frangi vesselness filter.
+    .. [3] http://mplab.ucsd.edu/tutorials/gabor.pdf.
+    """
+
+    (filtered_array, lambdas_array) = _frangi_hessian_common_filter(
+                                      image, scale, scale_step, beta1, beta2)
+
+    for i in range(len(filtered_array)):
+        filtered = filtered_array[i]
+        lambda1 = lambdas_array[i]
+        if black_ridges:
+            filtered[lambda1 < 0] = 0
+        else:
+            filtered[lambda1 >= 0] = 0
+        filtered_array[i] = filtered
```

Comment:
You can actually just do it with arrays!

```python
filtered, lambdas = _frangi_hessian_common_filter(...)

if black_ridges:
    filtered[lambdas < 0] = 0
else:
    filtered[lambdas > 0] = 0

return np.max(filtered, axis=0)
```
Note also that there's no need to test for equality since you are already just setting to 0. So you can use greater-than, instead of greater-than-or-equal.

Same for the Hessian filter below.